calcom/cal.com
# Feat: Web3 Rainbowkit Integration

## What does this PR do?



Adds features requested here https://github.com/calcom/cal.com/issues/1682

- [x] Rainbowkit integration
- [x] Crypto logic moved from booking page to interfaces
- [x] Support for Polygon mainnet, Arbitrum, and Optimism in addition to Ethereum mainnet
- [x] Basic implementation of [Gates](https://github.com/calcom/cal.com/discussions/3297)

**Loom video**: https://www.loom.com/share/3ae514f81e654aa6ab3708df48a3acd3

**Environment**: Staging(main branch) / Production

## Type of change


## How should this be tested?

Test manually by installing a wallet supported by Rainbow (usually metamask is easiest). The following is briefly how I tested the feature:

1. Install the `Rainbow` app from the App Store on your local deployment
2. Create a new event filling out the blockchain and contract address. Make sure you own the token at the contract address.
<img width="742" alt="image" src="https://user-images.githubusercontent.com/8162609/188219656-cb243ef0-826a-464d-bc0e-bc2e55e39526.png">
3. Visit the event link and connect your wallet

## Before Merging

- [x] Merge https://github.com/calcom/cal.com/pull/4049
- [x] Set `INFURA_API_KEY` and/or `ALCHEMY_API_KEY` in `.env.appStore`
<img width="901" alt="Find your Alchemy API key" src="https://user-images.githubusercontent.com/8162609/187499278-e5f03c3f-b4a5-430a-9121-f30207802d4c.png">
<img width="932" alt="Find your Infura API key" src="https://user-images.githubusercontent.com/8162609/187499759-425b4fc4-621b-4753-b77f-257b5055408a.png">

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
